r/PACKAGING.md

```diff
@@ -0,0 +1,121 @@
+  entitled `[R] CRAN release version X.X.X` and copy this checklist to the issue
+  
+- [ ] Evaluate the status of any failing
+  [nightly tests and nightly packaging builds](https://lists.apache.org/list.html?builds@arrow.apache.org): these are run
+  against the master branch and not the release branch, so the results may be
+  slightly out of date. After the CRAN-specific release branch has been created,
+  we can run any or all tests again using a GitHub PR and crossbow.
+- [ ] Check [current CRAN check results](https://cran.rstudio.org/web/checks/check_results_arrow.html)
+- [ ] Ensure the contents of the README is accurate
+- [ ] Run `urlchecker::url_check()` on the R directory at the release candidate
+  commit. Ignore any errors with badges as they will be removed in the CRAN release branch.
+- [ ] [Polish NEWS](https://style.tidyverse.org/news.html#news-release)
+- [ ] Prepare tweet thread highlighting new features
+
+Make pull requests into the [autobrew](https://github.com/autobrew) and
+[rtools-packages](https://github.com/r-windows/rtools-packages) repositories
+used by the configure script on MacOS and Windows. These pull requests will
+use the release candidate as the source.
+
+- [ ] Pull request to modify 
+  [the apache-arrow autobrew formula]( https://github.com/autobrew/homebrew-core/blob/master/Formula/apache-arrow.rb) 
+  to update the version, SHA, and any changes to dependencies and build steps that have changed in the
+  [copy of the formula we have of that formula in the Arrow repo](https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow.rb)
+- [ ] Pull request to modify
+  [the apache-arrow-static autobrew formula]( https://github.com/autobrew/homebrew-core/blob/master/Formula/apache-arrow-static.rb)
+  to update the version, SHA, and any changes to dependencies and build steps that have changed in the
+  [copy of the formula we have of that formula in the Arrow repo](https://github.com/apache/arrow/blob/master/dev/tasks/homebrew-formulae/autobrew/apache-arrow-static.rb)
+- [ ] Pull request to modify the 
+  [autobrew script](https://github.com/autobrew/scripts/blob/master/apache-arrow)
+  to include any additions made to
+  [r/tools/autobrew](https://github.com/apache/arrow/blob/master/r/tools/autobrew).
+- [ ] Pull request to modify the
+  [PKGBUILD script](https://github.com/r-windows/rtools-packages/blob/master/mingw-w64-arrow/PKGBUILD)
+  to reflect changes in
+  [ci/PKGBUILD](https://github.com/apache/arrow/blob/master/ci/scripts/PKGBUILD),
+  uncommenting the line that says "uncomment to test the rc".
+
+Prepare and check the .tar.gz that will be released to CRAN.
+
+- [ ] `git fetch upstream && git checkout release-X.X.X-rcXX`
+- [ ] `make build` (copies Arrow C++ into tools/cpp, prunes some unnecessary
+  components, and runs `R CMD build`)
+- [ ] `devtools::check("arrow_X.X.X.tar.gz")` locally
+- [ ] Run reverse dependency checks
```

Comment:
How?

Reply:
I have removed the crossbow job because it would have been a bunch of work to fix the timeout. But I was wondering why revdepcheck was taking SO LONG (even if you take the 2 core gha machines into account). And locally it will still timeout for a different reason ({targets} timesout with multiple workers??).

But I  would be fine with reworking the job to use your gist or even the build in function I did not know exists: https://www.r-bloggers.com/2019/04/checking-reverse-dependencies-the-tiny-way/ current docs: https://stat.ethz.ch/R-manual/R-patched/library/tools/html/check_packages_in_dir.html

Reply:
Doing it locally is a pain as I'm having to install a whole load of system libraries I don't need, to be able to run the gist; please can we rework the crossbow job?